test/test_framework.py

```diff
@@ -612,6 +612,63 @@ class CustomObject:
 
             framework_copy.commit()
 
+    def test_set_operations(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+            def __init__(self, framework, key):
+                super().__init__(framework, key)
+
+        # (input, operation, (a, b) result, (b, a) result
+        test_operations = [(
+            {"1"},
+            lambda a, b: a | b,
+            {"1", "a", "b"},
+            {"1", "a", "b"}
+        ), (
+            {"a", "c"},
+            lambda a, b: a - b,
+            {"b"},
+            {"c"}
+        ), (
+            {"a", "c"},
+            lambda a, b: a & b,
+            {"a"},
+            {"a"}
+        ), (
+            {"a", "c", "d"},
+            lambda a, b: a ^ b,
+            {"b", "c", "d"},
+            {"b", "c", "d"}
+        ), (
+            set(),
+            lambda a, b: set(a),
+            {"a", "b"},
+            set()
+        )]
+
+        framework = self.create_framework()
+
+        # Validate that operations between StoredSet and built-in sets only result in built-in sets being returned.
+        # Make sure that commutativity is preserved and that the original sets are not changed or used as a result.
+        for input, operation, ab_res, ba_res in test_operations:
```

Comment:
I led you astray here; `input` is a built-in and we shouldn't really be using it as a variable name.